src/lib/Libpython/pbs_python_external.c

```diff
@@ -195,10 +202,44 @@ pbs_python_ext_start_interpreter(
 		goto ERROR_EXIT;
 	}
 
+#ifndef WIN32
+	/*
+	 * Temporary set SIGINT to SIG_DFL, so Py_InitializeEx can setup proper SIGINT handler
+	 * see https://github.com/python/cpython/blob/3.6/Modules/signalmodule.c#L1280
+	 * as per above, If SIGINT is not set to SIG_DFL then Python won't register it's SIGINT handler
+	 * Which means Python won't raise KeyBoardInterrupt on PyErr_SetInterrupt()
+	 * instead it will throw NoneType not callable error
+	 */
+	sigemptyset(&act.sa_mask);
+	act.sa_flags   = 0;
+	act.sa_handler = SIG_DFL;
+	if (sigaction(SIGINT, &act, &oact) != 0) {
+		log_err(errno, __func__, "Failed to set SIG_DFL on SIGINT");
+		return 1;
+	}
+#else
+	oact = signal(SIGINT, SIG_DFL);
+	if (oact == SIG_ERR) {
+		log_err(errno, __func__, "Failed to set SIG_DFL on SIGINT");
+		return 1;
+	}
+#endif
+	/*
+	 * arg '1' means to not skip init of signals
+	 * we want signals to propagate to the executing
+	 * Python script to be able to interrupt it
+	 */
+	Py_InitializeEx(1);
 
-	Py_InitializeEx(1);  /* arg '1' means to not skip init of signals -    */
-	/* we want signals to propagate to the executing  */
-	/* Python script to be able to interrupt it       */
+	/* revert SIGINT to original sig handler */
+#ifndef WIN32
+	if (sigaction(SIGINT, &oact, NULL) != 0) {
+#else
+	if (signal(SIGINT, oact) == SIG_ERR) {
```

Comment:
My only suggestion would be to place the open brace outside of the CPP directive so there is only one open brace. Otherwise, some editors get confused when they see two open braces and one closing brace.